Owncloud iOs Client/AppDelegate.m

```diff
@@ -626,22 +627,16 @@ + (OCCommunication*)sharedOCCommunication
             return NSURLSessionAuthChallengePerformDefaultHandling;
         }];
         
-       /* [uploadSessionManager.reachabilityManager setReachabilityStatusChangeBlock:^(AFNetworkReachabilityStatus status) {
-            if (status == AFNetworkReachabilityStatusNotReachable) {
-                [[NSNotificationCenter defaultCenter] postNotificationName: NotReachableNetworkForUploadsNotification object: nil];
-                [[NSNotificationCenter defaultCenter] postNotificationName: NotReachableNetworkForDownloadsNotification object: nil];
-            }
-            
-        }];
-        
-        [uploadSessionManager.reachabilityManager startMonitoring];*/
-        
         NSURLSessionConfiguration *configurationDownload = nil;
         
-        if (IS_IOS8 || IS_IOS9) {
-            configurationDownload = [NSURLSessionConfiguration backgroundSessionConfigurationWithIdentifier:k_download_session_name];
-        }else{
-            configurationDownload = [NSURLSessionConfiguration backgroundSessionConfiguration:k_download_session_name];
+        if (k_is_sso_active || !k_is_background_active) {
+            configurationDownload = [NSURLSessionConfiguration defaultSessionConfiguration];
+        } else {
+            if (IS_IOS8 || IS_IOS9) {
```

Comment:
We could remove this if and use for all cases ``` backgroundSessionConfigurationWithIdentifier: ``` instead of ```backgroundSessionConfiguration``` that is deprecated.